### ISSUE TITLE
Youtube vids not working 

### ISSUE THREAD

**Issue by ceyhun-yildiz** [elementor/elementor#14135]
Hi

With the last update there has been a javascript error for displaying youtube vids on elementor pages. Using elementor free version, it was working before on this page but now it is not , please check developer tools on https://www.funkypaw.com/pawky-tv/

I have fixed the issue for our website but you should check and update it.

Thank you

--

**Comment by shilo-ey** [comment]
Hey, thanks for reporting.


The page you sent is blocked by Maintnance mode. can you please open it for inspection?

**Comment by Mitchy25** [comment]
I also had this issue so I rolled back to V3.1.1. 

The errors in the console are:

```
jQuery.Deferred exception: Cannot read property 'url' of undefined TypeError: Cannot read property 'url' of undefined
    at Video.onInit (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/preloaded-elements-handlers.min.js?ver=3.1.3:2:26586)
    at Video.Module.trigger (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:44357)
    at Video.e.each.n.<computed> [as trigger] (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:42981)
    at init (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:43142)
    at Video.Module (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:44435)
    at Video.child [as constructor] (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:44668)
    at new child (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:44668)
    at Video._createSuperInternal (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.1.3:2:1674)
    at new Video (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/preloaded-elements-handlers.min.js?ver=3.1.3:2:23944)
    at e.exports.addHandler (https://poweredbychange.com/wp-content/plugins/elementor/assets/js/frontend.min.js?ver=3.1.3:2:13634) undefined
```

```
Uncaught TypeError: Cannot read property 'url' of undefined
    at Video.onInit (preloaded-elements-handlers.min.js?ver=3.1.3:2)
    at Video.Module.trigger (frontend-modules.min.js?ver=3.1.3:2)
    at Video.e.each.n.<computed> [as trigger] (frontend-modules.min.js?ver=3.1.3:2)
    at init (frontend-modules.min.js?ver=3.1.3:2)
    at Video.Module (frontend-modules.min.js?ver=3.1.3:2)
    at Video.child [as constructor] (frontend-modules.min.js?ver=3.1.3:2)
    at new child (frontend-modules.min.js?ver=3.1.3:2)
    at Video._createSuperInternal (frontend-modules.min.js?ver=3.1.3:2)
    at new Video (preloaded-elements-handlers.min.js?ver=3.1.3:2)
    at e.exports.addHandler (frontend.min.js?ver=3.1.3:2)
```

```
common.min.js?ver=3.1.3:2 Uncaught ReferenceError: originalEvent is not defined
    at onError (common.min.js?ver=3.1.3:2)
    at dispatch (jquery.min.js?ver=3.5.1:2)
    at v.handle (jquery.min.js?ver=3.5.1:2)
```

**Comment by ceyhun-yildiz** [comment]
Hi, the issue still persists on free version of Elementor plugin with the same error as Mitchy identified

**Comment by ceyhun-yildiz** [comment]
This issue is still not resolved

**Comment by addMitt** [comment]
I believe I'm having this issue, albeit with some slightly different errors.

With no caching plugins or anything enabled, it seems like when a widget requires a JavaScript file to work, that file does not get included when viewing the page (unless you are, strangely, logged in to the backend as an administrator).

The YouTube part of the DOM element never shows up:

```
<div class="elementor-element elementor-element-cf4b777 elementor-aspect-ratio-169 elementor-widget elementor-widget-video" data-id="cf4b777" data-element_type="widget" data-settings="{&quot;youtube_url&quot;:&quot;https:\/\/www.youtube.com\/watch?v=wOwy_4RPkoo&quot;,&quot;video_type&quot;:&quot;youtube&quot;,&quot;controls&quot;:&quot;yes&quot;,&quot;aspect_ratio&quot;:&quot;169&quot;}" data-widget_type="video.default">
  <div class="elementor-widget-container">
    <div class="elementor-wrapper elementor-fit-aspect-ratio elementor-open-inline">
      <div class="elementor-video"></div>
    </div>
  </div>
</div>
```

The console reports an error related to a variable it requires not being defined (because the JavaScript file didn't load for it)

> Uncaught ReferenceError: elementorModules is not defined

And by comparing when logged in to the admin panel, I found that this is the file that is not being loaded:

> /wp/wp-content/plugins/elementor/assets/js/video.2be65474f4a78954309a.bundle.min.js

This does however, happen for any video site included with the Video widget (as the JavaScript for the entire widget is not being loaded).

**Comment by ScottTravisHartley** [comment]
If you're having this issue does it still occur if you add the following <script> tag before the </body> tag  


`<script src="https://www.youtube.com/iframe_api"> </script>`

I had a site where this something was occurring it took a few hours to see what was going on but the iframe_api JS was never being called adding the above src to the footer ended up allowing the videos to render.

**Comment by Aviad-Herman** [comment]
Hi,
Does this issue still occur on the latest versions of Elementor?

We need more info to replicate the issue.

1. Please add exact steps to reproduce (or preferably a short video, Loom is a great tool for this.)
   -Add as much details as possible
   -Even what user permissions you are using to edit the page (admin?)
   -Are you using the video widget to embed the video? HTML widget? Text widget? Something else?
   -Are you editing a page or template?
2. Disable all other plugins except Elementor
3. Use Elementor Hello theme
4. Turn of all Elementor Experiments (under Elementor-->Settings)
5. Clear your cache
6. Send us your system info


Without this I'm afraid there is not much we can do.

**Comment by ceyhun-yildiz** [comment]
The issue is fixed in the latest version of Elementor, thank you
